### PR TITLE
Upgrade Ember CLI to 3.12 LTS

### PR DESCRIPTION
**Pull request by SergeAstapov**
Continuation of #120, should unblock #123

### CODE REVIEW

**Comment by SergeAstapov** [comment]
@MelSumner @rwjblue #120 seem to be complete, please let me know if anything else left to get this in

**Comment by rwjblue** [comment]
CI seems unhappy, mind taking a look?

**Comment by SergeAstapov** [comment]
looking into CI issues and push updates

File: .travis.yml [hunk 1]
```diff
@@ -21,14 +21,20 @@ env:
   matrix:
     # we recommend new addons test the current and previous LTS
     # as well as latest stable release (bonus points to beta/canary)
-    - stage: "Additional Tests"
-      env: EMBER_TRY_SCENARIO=ember-lts-2.18
-    - env: EMBER_TRY_SCENARIO=ember-lts-3.4
-    - env: EMBER_TRY_SCENARIO=ember-lts-3.8
-    - env: EMBER_TRY_SCENARIO=ember-release
-    - env: EMBER_TRY_SCENARIO=ember-beta
+    - EMBER_TRY_SCENARIO=ember-lts-2.12
```
Comment (rwjblue) [inline .travis.yml]: I think the config/ember-try.js doesn't include 2.12 or 2.16 scenarios

Comment (SergeAstapov) [reply .travis.yml]: fixed, thank you @rwjblue !

File: README.md [hunk 2]
```diff
@@ -13,7 +13,17 @@ workflow by running during a component's `didRender` phase in non-production
 environments. This gives you instant feedback on if your components are
 accessible in any given state.
 
-## Installation
+
+Compatibility
+------------------------------------------------------------------------------
+
+* Ember.js v1.13 or above
```
Comment (rwjblue) [inline README.md]: We should change this to 2.18 (since that is all we are testing)

Comment (SergeAstapov) [inline README.md]: @rwjblue should this be 2.18 to match what ember-try config is?

File: package.json [hunk 3]
```diff
@@ -9,57 +9,63 @@
...
+    "ember-try": "^1.0.0",
+    "eslint-plugin-ember": "^6.2.0",
+    "eslint-plugin-node": "^9.0.1",
     "loader.js": "^4.7.0",
+    "qunit-dom": "^0.8.4",
     "sass": "^1.17.3"
   },
   "engines": {
-    "node": ">= 4"
+    "node": "8.* || >= 10.*"
```
Comment (rwjblue) [inline package.json]:
```suggestion
    "node": "8.* || 10.* || >= 12"
```

Comment (SergeAstapov) [reply package.json]: fixed

File: README.md (hunk 2 above)
Comment (SergeAstapov) [reply README.md]: fixed

**Comment by rwjblue** [comment]
> could you please advice if we still need registerAsyncHelper() in `addon-test-support/audit.js` for "older"/global tests helpers support?

Ya, I'd say not. Since this PR will already need to be a major bump, we can drop that helper also.

**Comment by SergeAstapov** [comment]
> > could you please advice if we still need registerAsyncHelper() in `addon-test-support/audit.js` for "older"/global tests helpers support?
> 
> Ya, I'd say not. Since this PR will already need to be a major bump, we can drop that helper also.

thanks for confirming, `registerAsyncHelper()` removed

**Comment by rwjblue** [comment]
@drewlee - I think you've been handling releases lately? Just wanted to point out that this one will require a major version bump (dropping Node < 8, Ember < 2.18, ember-cli < 2.13).

**Comment by drewlee** [comment]
@rwjblue Yeah, I can merge this PR and publish a new version this week. Thanks for the heads up.

**Pull request merged**